.github/workflows/build_optimized.yaml

```diff
@@ -0,0 +1,190 @@
+name: Smart Contracts
+run-name: "Building smart contracts, triggered by ${{ github.actor }}"
+
+on:
+  workflow_call:
+
+env:
+  CARGO_TERM_COLOR: "always"
+  # Release version is set to `ci` to allow building during checking and testing
+  # pipelines, which is ignored during optimized builds.
+  RELEASE_VERSION: "ci"
+  # Network's name is set to `main` to allow building during checking and testing
+  # pipelines, which is ignored during optimized builds.
+  NET_NAME: "main"
+
+jobs:
+  preparation:
+    runs-on: "ubuntu-latest"
+    steps:
+      - if: "github.event_name == 'workflow_dispatch'"
+        shell: "sh"
+        run: 'echo "# Run parameters:" >> "${GITHUB_STEP_SUMMARY}"'
+      - if: "fromJSON(inputs.perform_checks)"
+        shell: "sh"
+        run: 'echo "## Perform checks" >> "${GITHUB_STEP_SUMMARY}"'
+      - if: "fromJSON(inputs.optimized_build)"
+        shell: "sh"
+        run: 'echo "## Building optimized binaries" >> "${GITHUB_STEP_SUMMARY}"'
+  check_formatting:
+    runs-on: "ubuntu-latest"
+    needs:
+      - "preparation"
+    steps:
+      - uses: "actions/checkout@v3"
+        if: "fromJSON(inputs.perform_checks)"
+      - uses: "Nolus-Protocol/rust-ci-actions/rust_cache@v1"
+        if: "fromJSON(inputs.perform_checks)"
+      - uses: "Nolus-Protocol/rust-ci-actions/check_formatting@v1"
+        if: "fromJSON(inputs.perform_checks)"
+  check_codebase:
+    runs-on: "ubuntu-latest"
+    needs:
+      - "preparation"
+    steps:
+      - uses: "actions/checkout@v3"
+        if: "fromJSON(inputs.perform_checks)"
+      - uses: "Nolus-Protocol/rust-ci-actions/rust_cache@v1"
+        if: "fromJSON(inputs.perform_checks)"
+      - uses: "Nolus-Protocol/rust-ci-actions/check_codebase@v1"
+        if: "fromJSON(inputs.perform_checks)"
+  audit_dependencies:
+    runs-on: "ubuntu-latest"
+    needs:
+      - "preparation"
+    steps:
+      - uses: "actions/checkout@v3"
+        if: "fromJSON(inputs.perform_checks)"
+      - uses: "Nolus-Protocol/rust-ci-actions/rust_cache@v1"
+        if: "fromJSON(inputs.perform_checks)"
+      - uses: "Nolus-Protocol/rust-ci-actions/audit_dependencies@v1"
+        if: "fromJSON(inputs.perform_checks)"
+  linting:
+    runs-on: "ubuntu-latest"
+    needs:
+      - "check_formatting"
+      - "check_codebase"
+      - "audit_dependencies"
+    steps:
+      - uses: "actions/checkout@v3"
+        if: "github.event_name != 'workflow_dispatch' || !fromJSON(github.event.inputs.skip_checks)"
```

Comment:
it looks like the last refactoring did not reduce the repetitions of this line?

Reply:
It's still a bit of a mess and it's not ready quite yet.